src/components/SendFlow/Beacon/OriginationOperationSignModal.test.tsx

```diff
@@ -0,0 +1,93 @@
+import { BeaconMessageType, NetworkType, OperationRequestOutput } from "@airgap/beacon-wallet";
+import type { BatchWalletOperation } from "@taquito/taquito/dist/types/wallet/batch-operation";
+import BigNumber from "bignumber.js";
+
+import { OriginationOperationSignModal } from "./OriginationOperationSignModal";
+import { mockContractOrigination, mockImplicitAccount } from "../../../mocks/factories";
+import {
+  act,
+  dynamicModalContextMock,
+  render,
+  screen,
+  userEvent,
+  waitFor,
+} from "../../../mocks/testUtils";
+import { GHOSTNET, MAINNET } from "../../../types/Network";
+import { WalletClient } from "../../../utils/beacon/WalletClient";
+import { prettyTezAmount } from "../../../utils/format";
+import { useGetSecretKey } from "../../../utils/hooks/getAccountDataHooks";
+import { networksActions } from "../../../utils/redux/slices/networks";
+import { store } from "../../../utils/redux/store";
+import { executeOperations, makeToolkit } from "../../../utils/tezos";
+import { SuccessStep } from "../SuccessStep";
+
+const user = userEvent.setup();
+
+const message = {
+  id: "messageid",
+  type: BeaconMessageType.OperationRequest,
+  network: { type: NetworkType.GHOSTNET },
+  appMetadata: {},
+} as OperationRequestOutput;
+const operation = {
+  type: "implicit" as const,
+  sender: mockImplicitAccount(0),
+  signer: mockImplicitAccount(0),
+  operations: [mockContractOrigination(0)],
+};
+const fee = BigNumber(123);
+
+jest.mock("../../../utils/tezos", () => ({
+  ...jest.requireActual("../../../utils/tezos"),
+  executeOperations: jest.fn(),
+  makeToolkit: jest.fn(),
+}));
+
+jest.mock("../../../utils/hooks/getAccountDataHooks", () => ({
+  ...jest.requireActual("../../../utils/hooks/getAccountDataHooks"),
+  useGetSecretKey: jest.fn(),
+}));
+
+describe("<OriginationOperationSignModal />", () => {
+  it("renders fee", () => {
+    render(<OriginationOperationSignModal fee={fee} message={message} operation={operation} />);
+
+    expect(screen.getByText(prettyTezAmount(fee))).toBeVisible();
+  });
+
+  it("uses correct network", async () => {
```

Comment:
I think it's enough to check that when you click the button `executeOperations` gets the correct payload, and the `WalletClient` too. 
this test was written in another component just to make sure that the network selection works correctly, no need to copy it here